articles/getting-started-with-mariadb-using-docker-and-nodejs/index.md

```diff
@@ -0,0 +1,180 @@
+
+```bash
+$ docker run -p 127.0.0.1:3306:3306  --name mariadbdemo -e MARIADB_ROOT_PASSWORD=Demo123@ -d mariadb:latest
+```
+
+The above command will download and create a MariaDB server container to connect to and communicate using the MariaDB client.
+
+The container's name is `mariadbdemo`, and the `Demo123@` is the password for the MariaDB root user. 
+
+Note that other SQL clients can be used, but we will use MariaDB's latest version to keep things simple and uniform for this guide.
+
+Next, will be to connect the MariaDB instance by running the below command in the terminal:
+
+```bash
+$ mariadb --host 127.0.0.1 -P 3306 --user root –pDemo123@
+```
+
+The below screen will appear, meaning that we have connected to the MariaDB instance successfully.
+
+![maria-db-root](/engineering-education/getting-started-with-mariadb-using-docker-and-nodejs/maria-db-root.PNG)
+
+The next step will be to create a new database by running the below command:
+
+```sql
+CREATE DATABASE demodb;
+```
+
+Then we create a new table by running the below command:
+
+```sql
+CREATE TABLE demodb.persons (Name VARCHAR(50));
+```
+
+Finally, will be to insert several records as below:
+
+```sql
+INSERT INTO demodb.persons VALUES ('Bob'), ('Alice'), ('Ben'), ('Mary');
+```
+
+### Connecting to MariaDB with Node.js
+With MariaDB database downloaded and installed. It is now ready to use within a new Node.js app.
+
+To begin, we create a new directory and then browse into it. Next, create a new JavaScript file named `server.js` which will act as the main module of the Node server. 
+
+Next will be to browse to the directory and execute the following command in the terminal:
+
+```bash
+$ npm init
+```
+
+One can fill out the prompts that appear or leave them as default by pressing the enter key through all the options.
+
+A `package.json` file will be auto-generated in the same directory as the `server.js` file.
+
+Note that we now have a runnable Node application.
+
+Next, we will install an [express](https://expressjs.com/) package, acting as our web framework for the Node application. To install the express package, we execute the below command.
+
+```bash
+$ npm install express
+```
+
+Then next, we will install a [MariaDB Node.js connector](https://mariadb.com/kb/en/nodejs-connector/). It will establish a connection and communicate with our MariaDB instance.
+
+The below command will be used to install the MariaDB Node.js connector:
+
+```bash
+$ npm install mariadb
+```
+
+Now, we can add the code that will connect to MariaDB.
+
+The First will be to create a new reusable [module](https://nodejs.org/api/modules.html) file by the name `db.js`.
+
+The `db.js` module will utilize the MariaDB Node.js connector to enable our application to connect to and communicate with MariaDB.
+
+The code for the `db.js` will be as follows:
+
+```JavaScript
+// here we import the mariadb
+var mariadb = require('mariadb');
```

Comment:
Preferably use `const` as this variable isn't going to be changing in the duration of the app. 

```suggestion
const mariadb = require('mariadb');
```